resp/response_type.go

```diff
@@ -0,0 +1,284 @@
+package resp
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/xml"
+	"math"
+	"math/cmplx"
+)
+
+const LaplaceRadiansSecondPzTransferFunction = "LAPLACE (RADIANS/SECOND)"
+
+type Float struct {
+	Value float64 `xml:",chardata"`
+}
+
+type Units struct {
+	Name        string `xml:"Name"`
+	Description string `xml:"Description,omitempty"`
+}
+
+type ApproximationBound struct {
+	Value float64 `xml:",chardata"`
+}
+
+type NumeratorCoefficient struct {
+	I     int     `xml:"i,attr"`
+	Value float64 `xml:",chardata"`
+}
+
+type PolynomialCoefficient struct {
+	Number int     `xml:"number,attr"`
+	Value  float64 `xml:",chardata"`
+}
+
+type CoefficientNumerator struct {
+	Number int     `xml:"number,attr"`
+	Value  float64 `xml:",chardata"`
+}
+
+type CoefficientDenominator struct {
+	Number int     `xml:"number,attr"`
+	Value  float64 `xml:",chardata"`
+}
+
+type PoleZero struct {
+	Number int `xml:"number,attr"`
+
+	Real      Float `xml:"Real"`
+	Imaginary Float `xml:"Imaginary"`
+}
+
+type CoefficientsType struct {
+	ResourceId  string `xml:"resourceId,attr,omitempty"`
+	Name        string `xml:"name,attr,omitempty"`
+	Description string `xml:"description,attr,omitempty"`
+
+	InputUnits             Units  `xml:"InputUnits"`
+	OutputUnits            Units  `xml:"OutputUnits"`
+	CfTransferFunctionType string `xml:"CfTransferFunctionType"`
+
+	Numerators   []CoefficientNumerator   `xml:"Numerator,omitempty"`
+	Denominators []CoefficientDenominator `xml:"Denominator,omitempty"`
+}
+
+type DecimationType struct {
+	InputSampleRate float64 `xml:"InputSampleRate"`
+	Factor          int     `xml:"Factor"`
+	Offset          int     `xml:"Offset"`
+	Delay           float64 `xml:"Delay"`
+	Correction      float64 `xml:"Correction"`
+}
+
+type FirType struct {
+	ResourceId  string `xml:"resourceId,attr"`
+	Name        string `xml:"name,attr,omitempty"`
+	Description string `xml:"description,attr,omitempty"`
+
+	InputUnits  Units  `xml:"InputUnits"`
+	OutputUnits Units  `xml:"OutputUnits"`
+	Symmetry    string `xml:"Symmetry"`
+
+	NumeratorCoefficients []NumeratorCoefficient `xml:"NumeratorCoefficient"`
+}
+
+type PolesZerosType struct {
+	ResourceId  string `xml:"resourceId,attr"`
+	Name        string `xml:"name,attr,omitempty"`
+	Description string `xml:"description,attr,omitempty"`
+
+	InputUnits  Units `xml:"InputUnits"`
+	OutputUnits Units `xml:"OutputUnits"`
+
+	PzTransferFunctionType string  `xml:"PzTransferFunctionType"`
+	NormalizationFactor    float64 `xml:"NormalizationFactor"`
+	NormalizationFrequency float64 `xml:"NormalizationFrequency"`
+
+	Zeros []PoleZero `xml:"Zero"`
+	Poles []PoleZero `xml:"Pole"`
+}
+
+// Gain ccalculates the poles and zeros response gain at a given frequency
+func (pz PolesZerosType) Gain(freq float64) float64 {
+
+	var w complex128
+	switch pz.PzTransferFunctionType {
```

Comment:
nitpick(non-blocking): I believe `switch` is recommend for 3 or more conditions, and `if` for simply binary.

Reply:
This is my c programming background protesting about the lack of a conditional statement in go.  I've tended to have an aversion to if else blocks and have standardised on a switch.

Reply:
I understand :-) I didn't realise how much I missed conditional operators until I started using Go...